nuxeo-core/nuxeo-core-mimetype/src/main/java/org/nuxeo/ecm/platform/mimetype/service/MimetypeRegistryService.java

```diff
@@ -385,45 +376,60 @@ public String getMimetypeFromBlobWithDefault(Blob blob, String defaultMimetype)
         }
     }
 
-    /**
-     * Finds the mimetype of some content according to its filename and / or binary content.
-     *
-     * @param filename extension to analyze
-     * @param blob content to be analyzed if filename is ambiguous
-     * @param defaultMimetype defaultMimeType to be used if no found
-     * @return the string mimetype
-     * @throws MimetypeDetectionException
-     * @author lgodard
-     */
     @Override
     public String getMimetypeFromFilenameAndBlobWithDefault(String filename, Blob blob, String defaultMimetype)
             throws MimetypeDetectionException {
         try {
             return getMimetypeFromFilename(filename);
         } catch (MimetypeNotFoundException e) {
-            // failed to detect mimetype on extension: fallback to Blob based
-            // detection
-            try {
-                return getMimetypeFromBlob(blob);
-            } catch (MimetypeNotFoundException mtnfe) {
-                return defaultMimetype;
+            // failed to detect mimetype on extension:
+            // fallback to calculate mimetype from blob content
+            return getMimetypeFromBlobWithDefault(blob, defaultMimetype);
+        }
+    }
+
+    @Override
+    public String getMimetypeFromFilenameWithBlobMimetypeFallback(String filename, Blob blob, String defaultMimetype)
+            throws MimetypeDetectionException {
+        try {
+            return getMimetypeFromFilename(filename);
+        } catch (MimetypeNotFoundException e) {
+            // failed to detect mimetype on extension:
+            // fallback to the blob defined mimetype
+            String mimeTypeName = blob.getMimeType();
+            MimetypeEntry mimetypeEntry = getMimetypeEntryByName(mimeTypeName);
+            if (mimetypeEntry != null) {
+                return mimeTypeName;
+            } else {
+                // failed to detect mimetype on blob:
+                // fallback to calculate mimetype from blob content
+                return getMimetypeFromBlobWithDefault(blob, defaultMimetype);
```

Comment:
Could be:

            // failed to detect mimetype on extension:
            // fallback to the blob defined mimetype
            String mimeTypeName = blob.getMimeType();
            if (isMimetypeEntry(mimeTypeName)) {
                return mimeTypeName;
            }
            // failed to detect mimetype on blob:
            // fallback to calculate mimetype from blob content
            return getMimetypeFromBlobWithDefault(blob, defaultMimetype);

with

    protected boolean isMimetypeEntry(String name) {
        return mimetypeByNormalisedRegistry.containsKey(name);
    }